### ISSUE TITLE
undefined config when running under jest

### ISSUE THREAD

**Issue by codyzu** [krakenjs/confit#67]
The code segment below fails for me when running tests that use confit under [jest](https://facebook.github.io/jest/). However, executing the code directly (not using jest) works fine. Jest must muck with the global Object? Not really sure, but I think this code is too fragile.

If/when I find some time, I can try to create an example project that shows the problem. Regardless, my feeling is the code below does not pass the code smell test.

The result is that my tests that use confit return `undefined` for any config that is accessed with colon delimiters. The same code runs fine outside of jest.

https://github.com/krakenjs/confit/blob/2.x/lib/config.js#L34

```javascript
                if (obj.constructor !== Object) {
                    // Do not allow traversal into complex types,
                    // such as Buffer, Date, etc. So, this type
                    // of key will fail: 'foo:mystring:length'
                    return undefined;
                }
```

node 7.2.1
jest 17.0.3
confit 2.1.0

**Comment by codyzu** [comment]
This seems to be resolved by upgrading jest to version 19.0.2.

Feel free to close the issue... although I'm still not sure if the code snippet above passes the code smell test :-)

**Comment by shaunwarman** [comment]
Thanks for the update @codyzu - I agree, @totherik does smell
